src/Evolution/Systems/ScalarTensor/Sources/ScalarSource.cpp

```diff
@@ -0,0 +1,25 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#include "Evolution/Systems/ScalarTensor/Sources/ScalarSource.hpp"
+
+#include "DataStructures/Tensor/Tensor.hpp"
+#include "Utilities/ConstantExpressions.hpp"
+#include "Utilities/Gsl.hpp"
+
+namespace ScalarTensor {
+
+void add_scalar_source_to_dt_pi_scalar(
+    gsl::not_null<Scalar<DataVector>*> dt_pi_scalar,
+    const Scalar<DataVector>& scalar_source, const Scalar<DataVector>& lapse) {
+  dt_pi_scalar->get() += get(lapse) * scalar_source.get();
+}
+
+void compute_mass_source(
+    const gsl::not_null<Scalar<DataVector>*> scalar_source,
+    const Scalar<DataVector>& psi, const double mass_psi) {
+  // Make sure it has the same size
+  *scalar_source = make_with_value<Scalar<DataVector>>(psi, 0.);
+  scalar_source->get() = square(mass_psi) * psi.get();
```

Comment:
use `get(*scalar_source)` and `get(psi)`

Reply:
Done